### PR TITLE
ddl: disallow dropping index on auto_increment column

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve? 
```sql
drop table if exists t;
create table t (a int auto_increment, unique key (a));
alter table t drop index a;
```

TiDB:
```
Query OK, 0 rows affected
Time: 0.003s

Query OK, 0 rows affected
Time: 0.013s

Query OK, 0 rows affected
Time: 0.025s
```

MySQL 5.7:
```
Query OK, 0 rows affected
Time: 0.004s

Query OK, 0 rows affected
Time: 0.248s

(1075, 'Incorrect table definition; there can be only one auto column and it must be defined as a key')
```

### What is changed and how it works?

When `DropIndex()`, check `AutoIncrement` flag in indexed columns.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: planner/core/preprocess.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ package core
 
 import (
 	"fmt"
+	"github.com/pingcap/tidb/meta/autoid"
```
Comment (zimulala) [inline planner/core/preprocess.go]: Please update the order of imported package.

Comment (tangenta) [reply planner/core/preprocess.go]: Updated. PTAL :)

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -3263,7 +3263,8 @@ func (d *ddl) DropIndex(ctx sessionctx.Context, ti ast.Ident, indexName model.CI
 		return errors.Trace(infoschema.ErrTableNotExists.GenWithStackByArgs(ti.Schema, ti.Name))
 	}
 
-	if indexInfo := t.Meta().FindIndexByName(indexName.L); indexInfo == nil {
+	var indexInfo *model.IndexInfo
```
Comment (ngaut) [inline ddl/ddl_api.go]: indexInfo := t.Meta().FindIndexByName(indexName.L)

Comment (tangenta) [reply ddl/ddl_api.go]: Updated. PTAL

File: ddl/ddl_api.go [hunk 3]
```diff
@@ -3272,6 +3273,13 @@ func (d *ddl) DropIndex(ctx sessionctx.Context, ti ast.Ident, indexName model.CI
 		return err
 	}
 
+	cols := t.Cols()
+	for _, idxCol := range indexInfo.Columns {
+		if mysql.HasAutoIncrementFlag(cols[idxCol.Offset].Flag) {
+			return errors.Trace(autoid.ErrWrongAutoKey)
```
Comment (ngaut) [inline ddl/ddl_api.go]: No need to trace, this is the only place to return ErrWrongAutoKey.

**Comment by ngaut** [comment]
LGTM

**Pull request merged**